products/endpoints/backend/api.py

```diff
@@ -269,22 +399,200 @@ def update(self, request: Request, name=None, *args, **kwargs) -> Response:
             capture_exception(e)
             raise ValidationError("Failed to update endpoint.")
 
+    def _get_saved_query_context(self, request: Request) -> dict:
+        """Create context dict for DataWarehouseSavedQuerySerializer."""
+        database = create_hogql_database(team=self.team)
+        return {
+            "team_id": self.team.id,
+            "request": request,
+            "database": database,
+        }
+
+    def _get_or_create_saved_query(self, endpoint: Endpoint, request: Request) -> DataWarehouseSavedQuery:
+        """Get or create a saved query for an endpoint."""
+        saved_query_name = f"__endpoint_{self.team.id}_{endpoint.name}"
+        saved_query = DataWarehouseSavedQuery.objects.filter(team_id=self.team_id, name=saved_query_name).first()
+
+        if not saved_query:
+            context = self._get_saved_query_context(request)
+            serializer = DataWarehouseSavedQuerySerializer(
+                data={
+                    "name": saved_query_name,
+                    "query": endpoint.query,
+                },
+                context=context,
+            )
+            serializer.is_valid(raise_exception=True)
+            saved_query = serializer.save()
+
+        return saved_query
+
+    def _update_saved_query_sync_frequency(
+        self,
+        saved_query: DataWarehouseSavedQuery,
+        sync_frequency: DataWarehouseSyncInterval,
+        request: Request,
+    ) -> DataWarehouseSavedQuery:
+        """Update sync frequency on a saved query."""
+        context = self._get_saved_query_context(request)
+        request.data["sync_frequency"] = sync_frequency
+
+        serializer = DataWarehouseSavedQuerySerializer(
+            saved_query,
+            data={},
+            partial=True,
+            context=context,
+        )
+        serializer.is_valid(raise_exception=True)
+        return serializer.save()
+
+    def _enable_materialization(
```

Comment:
This won't actually materialize the saved query btw, there's a bunch of logic that needs to happen. Unfortunately, this logic isn't in a centralized place, but instead is replicated in 2 places, see:
- `posthog/warehouse/api/saved_query.py#update`
- `posthog/warehouse/models/datawarehouse_managed_viewset.py#sync_views`

if you could centralize this logic into `posthog/warehouse/models/datawarehouse_saved_query.py` (you'll see `revert_materialization` is already in there, for example), then you can just call this directly and not worry about all the logic in your API layer 

cc @rafaeelaudibert